### ISSUE TITLE
chap03_Linear_Regression_regression_formula2

### ISSUE THREAD

**Issue by hih0124** [hih0124/lecture_deep_learning#21]
"""
다중 선형 회귀 방정식 : 행렬곱 이용
- X(n) -> Y
- y_pred = X1 * a1 + X2 + a2 + ...
- y_pred = tf.matmul(X, a) + b
"""

import tensorflow as tf

- X,Y  변수 정의
X = [[1.0, 2.0]] # [1, 2] : 입력
Y = 2.5 # 정답

- a,b 변수 정의 : 수정가능(난수 상수)
a = tf.Variable(tf.random.normal([2, 1])) 
-기울기 : [2, 1] # 입력과 수일치 되어야 한다.
b = tf.Variable(tf.random.normal([1])) # 상수

- model 식 정의
y_pred = tf.math.add(tf.matmul(X,a), b) # tf.matmul(X,a)+ b
print(y_pred) # tf.Tensor([[-0.08399847]], shape=(1, 1), dtype=float32)
'''
tf.matmul(X,a) : 행렬곱
전제조건
1. X, a -> 행렬
2. X열수 == a 행수
'''
- model error
err = Y - y_pred

- loss function : 손실 반환
loss = tf.reduce_mean(tf.square(err))

print("최초 기울기(a)와 절편(b)")
print("a = {}, b = {}".format(a.numpy(), b.numpy()))

print("model error")
print("model error =", err.numpy()) # model error = [[6.5573015]]
print("loss =", loss.numpy()) # loss = 42.998203
